### PR TITLE
Feat: Api call

### PR DESCRIPTION
**Pull request by sri-rishi**
##Api call for getting data from server (it's just api call to get data) 

### files to check

- Index.js
- data.context.js
- server-request.js
- ProductDataList.jsx

### CODE REVIEW

File: src/data/sever-request.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+
+export const getProductFromMockServer = async(setData) => {
+    try {
+        const response = await axios.get("/api/products");
+        setData(response?.data?.products);
+    }catch(e) {
+        console.log(e);
```
Comment (yaswanthmyneni) [inline src/data/sever-request.js]: suggestion: use console.error(e) for displaying error messages

**Review by yaswanthmyneni — APPROVED** [review]
the code is clean, you can merge.

File: src/data/sever-request.js (hunk 1 above)
Comment (sri-rishi) [reply src/data/sever-request.js]: Done

File: src/pages/ProductList/components/ProductDataList/ProductDataList.jsx [hunk 2]
```diff
@@ -1,11 +1,12 @@
...
+import { useData } from "../../../../Context/data.context"
 
 export const ProductDataList = () => {
+    const {data} = useData()
     return (
         <main className="main-box grid grid-3-column">
             {
-                productData.map((productItems) => (
+                data.map((productItems) => (
                         <VerticalCard key={productItems._id} productItems={productItems}/>
```
Comment (bhtibrewal) [inline src/pages/ProductList/components/ProductDataList/ProductDataList.jsx:10]: Work on the naming, try more descriptive name as their will be log of data ahead

File: src/pages/ProductList/components/ProductDataList/ProductDataList.jsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import { VerticalCard } from "../../../../Components/VerticalCard/VerticalCard"
-import { productData } from "../../../../data/product.data"
+import { useData } from "../../../../Context/data.context"
```
Comment (bhtibrewal) [inline src/pages/ProductList/components/ProductDataList/ProductDataList.jsx:2]: Again import all in one index.js and then export

**Review by bhtibrewal — APPROVED** [review]
Good to merge, just a suggestion manage all data in one reducer

**Pull request merged**